src/single_node/main.rs

```diff
@@ -109,32 +85,27 @@ mod tests {
         assert!(reply.is_some());
         assert_eq!("v1".to_string(), reply.unwrap());
 
-        let reply = ClientCommand::Get {
-            key: "k1".to_string(),
-        }
-        .send_to(address)
-        .await
-        .unwrap();
-        assert!(reply.is_some());
-        assert_eq!("v1".to_string(), reply.unwrap());
-
-        let reply = ClientCommand::Set {
-            key: "k1".to_string(),
-            value: "v2".to_string(),
-        }
-        .send_to(address)
-        .await
-        .unwrap();
-        assert!(reply.is_some());
-        assert_eq!("v2".to_string(), reply.unwrap());
+        assert_eventually_equals(address, "k1", "v1").await;
+    }
 
-        let reply = ClientCommand::Get {
-            key: "k1".to_string(),
-        }
-        .send_to(address)
-        .await
-        .unwrap();
-        assert!(reply.is_some());
-        assert_eq!("v2".to_string(), reply.unwrap());
+    /// Send Get commands to the given address with delayed retries to give it time for a transaction
+    /// to propagate. Fails if the expected value isn't read after 20 seconds.
+    async fn assert_eventually_equals(address: SocketAddr, key: &str, value: &str) {
```

Comment:
if this is exactly the same as used in the other module, can you move it to a common test helper place?

(not sure what the convention is for test helpers, maybe @pablodeymo has recommendations)